test/e2e/operator.go

```diff
@@ -10,21 +10,49 @@ import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
+	"github.com/Azure/ARO-RP/pkg/util/ready"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/util/retry"
 
 	aro "github.com/Azure/ARO-RP/operator/apis/aro.openshift.io/v1alpha1"
 )
 
-var _ = Describe("Internetreachable shows correct condition", func() {
-	Specify("default list should all be reachable", func() {
+var pullSecretName = types.NamespacedName{Name: "pull-secret", Namespace: "openshift-config"}
+
+func pullSecretExists(namespace string, name string) (done bool, err error) {
+	_, err = Clients.Kubernetes.CoreV1().Secrets(pullSecretName.Namespace).Get(pullSecretName.Name, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	return true, nil
```

Comment:
This can be simplified:

```suggestion
	return err == nil, err
```